src/shared/schemas.ts

```diff
@@ -211,7 +215,9 @@ export async function getRemoteOrCachedFile(params: {
         showUrl: true,
         // updates curr version
         onSuccess: contents => {
-            writeFileSync(params.filepath, contents)
+            const parsedFile = JSON.parse(contents)
+            parsedFile.title = params.title
+            writeFileSync(params.filepath, JSON.stringify(parsedFile))
             params.extensionContext.globalState.update(params.cacheKey, params.version)
```

Comment:
I think all of these lines can fail :) But rather than handle it here, perhaps we should handle it in HttpResourceFetcher: https://github.com/aws/aws-toolkit-vscode/blob/acbd172d8e0de799e880ba83aff1f6a5562f11af/src/shared/resourcefetcher/httpResourceFetcher.ts#L87

Reply:
Yup they can fail though I'd say that's desirable here. Only thing that is non-critical would be updating the cache key.

`onSuccess` should probably be removed. I mean the resource fetchers in general should be removed but eh.